src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -954,8 +951,9 @@ export default class AircraftModel {
      */
     isOnGround() {
         const errorAllowanceInFeet = 5;
-        const airport = AirportController.airport_get();
-        const nearRunwayAltitude = abs(this.altitude - this.fms.currentRunway.elevation) < errorAllowanceInFeet;
+        const airport = this.fms.arrivalAirportModel || this.fms.departureAirportModel;
+        const runwayModel = this.fms.arrivalRunwayModel || this.fms.departureRunwayModel;
```

Comment:
these two lines make me nervous.  can we abstract this logic to the `fms` somehow?  How would we handle cases were both `arrivalRunwayModel` and `departureRunwayModel` are set in the `fms`?

Reply:
restructured:
```
isOnGround() {
    let airportModel = this.fms.departureAirportModel;
    let runwayModel = this.fms.departureRunwayModel;

    if (this.isArrival()) {
        airportModel = this.fms.arrivalAirportModel;
        runwayModel = this.fms.arrivalRunwayModel;
    }

    const errorAllowanceInFeet = 5;
    const nearRunwayAltitude = abs(this.altitude - runwayModel.elevation) < errorAllowanceInFeet;
    const nearAirportAltitude = abs(this.altitude - airportModel.elevation) < errorAllowanceInFeet;

    return nearRunwayAltitude || nearAirportAltitude;
}
```

Also, we definitely do NOT want a method for the `||`ing because that is a bad practice, we should always request either the arrival runway or departure runway. If we're asking for a runway assignment without that context, we're doing _something_ wrong.